src/rust/ide/ast/impl/src/crumbs.rs

```diff
@@ -461,6 +461,12 @@ pub trait Crumbable {
         let child = self.get(&crumb)?;
         Ok(Located::new(crumb,child))
     }
+
+    /// Enumerates all AST being a direct children of the given AST node.
+    fn direct_children<'a>(&'a self) -> Box<dyn Iterator<Item = ChildAst<'a>> + 'a> {
+        let iter = self.enumerate().map(|(crumb,ast)| ChildAst::new(crumb,ast));
+        Box::new(iter)
+    }
```

Comment:
I would name it just `children`. It is obvious that it means direct children. I would be very surprised if `children` method would give me a deep iterator. So rather lets use sophisticated name for deep-iteration instead.